lib/ts/recipe/multifactorauth/index.ts

```diff
@@ -0,0 +1,208 @@
+            requiredSecondaryFactorsForTenant: defaultMFARequirementsForTenant,
+            completedFactors,
+            userContext: ctx,
+        });
+    }
+
+    static async markFactorAsCompleteInSession(
+        session: SessionContainerInterface,
+        factorId: string,
+        userContext?: Record<string, any>
+    ) {
+        return Recipe.getInstanceOrThrowError().recipeInterfaceImpl.markFactorAsCompleteInSession({
+            session,
+            factorId,
+            userContext: getUserContext(userContext),
+        });
+    }
+
+    static async getFactorsSetupForUser(tenantId: string, userId: string, userContext?: Record<string, any>) {
+        const ctx = getUserContext(userContext);
+        const user = await getUser(userId, ctx);
+        if (!user) {
+            throw new Error("Unknown user id");
+        }
+
+        return Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getFactorsSetupForUser({
+            tenantId,
+            user,
+            userContext: ctx,
+        });
+    }
+
+    static async getRequiredSecondaryFactorsForUser(userId: string, userContext?: Record<string, any>) {
+        return Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getRequiredSecondaryFactorsForUser({
+            userId,
+            userContext: getUserContext(userContext),
+        });
+    }
+
+    static async addToRequiredSecondaryFactorsForUser(
+        userId: string,
+        factorId: string,
+        userContext?: Record<string, any>
+    ) {
+        await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.addToRequiredSecondaryFactorsForUser({
+            userId,
+            factorId,
+            userContext: getUserContext(userContext),
+        });
+    }
+
+    static async removeFromRequiredSecondaryFactorsForUser(
+        userId: string,
+        factorId: string,
+        userContext?: Record<string, any>
+    ) {
+        await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.removeFromRequiredSecondaryFactorsForUser({
+            userId,
+            factorId,
+            userContext: getUserContext(userContext),
+        });
+    }
+}
+
+export let init = Wrapper.init;
+
+export let assertAllowedToSetupFactorElseThrowInvalidClaimError =
+    Wrapper.assertAllowedToSetupFactorElseThrowInvalidClaimError;
+export let markFactorAsCompleteInSession = Wrapper.markFactorAsCompleteInSession;
+export let getFactorsSetupForUser = Wrapper.getFactorsSetupForUser;
+export let getRequiredSecondaryFactorsForUser = Wrapper.getRequiredSecondaryFactorsForUser;
+export const addToRequiredSecondaryFactorsForUser = Wrapper.addToRequiredSecondaryFactorsForUser;
+export const removeFromRequiredSecondaryFactorsForUser = Wrapper.removeFromRequiredSecondaryFactorsForUser;
+
+export const FactorIds = {
+    EMAILPASSWORD: "emailpassword",
+    OTP_EMAIL: "otp-email",
+    OTP_PHONE: "otp-phone",
+    LINK_EMAIL: "link-email",
+    LINK_PHONE: "link-phone",
+    THIRDPARTY: "thirdparty",
```

Comment:
```suggestion
```

Reply:
Done

Reply:
Actually, undo this change please, cause it can still be used in first factors.

Reply:
Undone